main.py

```diff
@@ -171,41 +175,45 @@ def update(self, circuit_id):
         try:
             evc = self.circuits[circuit_id]
         except KeyError:
-            result = {'response': f'circuit_id {circuit_id} not found'}
-            status = 404
+            result = f'circuit_id {circuit_id} not found'
+            log.debug('update result %s %s', result, 404)
+            raise NotFound(result)
+
+        if evc.archived:
+            result = f'Can\'t update archived EVC'
+            log.debug('update result %s %s', result, 405)
+            raise MethodNotAllowed(['GET'], result)
+
+        try:
+            data = request.get_json()
+        except BadRequest:
+            result = 'The request body is not a well-formed JSON.'
+            log.debug('update result %s %s', result, 400)
+            raise BadRequest(result)
+        if data is None:
+            result = 'The request body mimetype is not application/json.'
```

Comment:
Is this always right? What if the mime-type _is_ `application/json` but it is really coming empty?

Reply:
If the mime-type is correct but the body is empty, the request is invalid, Flask doesn't even call `create_circuit`, it raises an exception before that.